docs/UserGuide.md

```diff
@@ -183,152 +193,215 @@ Adds an event with specific starting and ending date or time to TaskTracker.
+
+* 'list due' will show all deadline according to their due time and due date. (when sorting
+the deadline without due time, the sorting will use 11:59 pm of the due date as deafault)
+
+* Every tasks (floating tasks / events / deadlines) will have a unique index to differentiate
+their uniqueness. And their index will also be listed besides them when `list` command is
+excuted.
+
+* The overdue/past/finished tasks will not be listed.
 
 ### Deleting a floating task/event/deadline: `del`
 
 Delete a useless floating task/event/deadline on TaskTracker.
 
-    del "TASK_NAME"
+    del TASK_UNIQUE_INDEX
 
-* Task name should be in a pair of quotation marks.
+* `TASK_UNIQUE_INDEX` will be different from each to each single task.
 
-* Here we assume that no duplicate task name is used among all the task.
+* A task's `TASK_UNIQUE_INDEX` will be the same as what is shown when `list` command is excuted.
+The user can refer `list` cammand to look for the index of a certain task.
+
+* `TASK_UNIQUE_INDEX` will never be changed once a index is assigned to a task when created.
 
 #### Examples
 
-* `del "project meeting"`
+* `list event`
+
+List all the events stored in the database with their unique index number. 
+
+`del 00123` <br>
 
-* `del "CS2102 week 8 lecture"`
+Delete the task with the unique index of `00123`.
 
-* `del "reading a book"`
+### Edit a floating task/event/dealine: `edit`
 
-### Edit a floating task: `edit`
+* Edit command can only edit the parameters of each commands but cannot transform a task to another
+type. For example, `eidt` cannot transform a floating task to a event task.
+
+#### Edit an floating task:
 
 Edit a floating task to revise its name or priority.
 
-    edit "TASK_NAME" [n-NEW_NAME | p-PRIORITY]...
+    edit TASK_UNIQUE_INDEX [n-NEW_NAME | p-PRIORITY]...
 
-* Task name should be in a pair of quotation marks.
+* Quotation marks are not necessary for `NEW_NAME`.
 
 * `PRIORITY` should only be the interger ranges from `0` to `5`.
 
-* Here we assume that no duplicate task name is used among all the task.
-
 #### Examples
 
-* `edit "washing clothes" p-0`
+* `edit 00124 p-0`
+
+Edit task with unique index of `00124`'s priority to 0.
 
-* `edit "buy textbooks" n-buy stationary`
+* `edit 00124 n-buy stationary`
 
-* `edit "go shopping" n-"go to Nanyang Mart" p-1`
+Edit task with unique index of `00124`'s name to `buy stationary`.
 
-### Edit an event : `edit`
+* `edit 00125 n-"go to Nanyang Mart" p-1`
+
+Edit task with unique index of `00125`'s name to `go to Nanyang Mart` and priority to 1.
+
+#### Edit an event :
 
 Edit an event to revise its name, starting/ending date/time and location.
 
-    edit "TASK_NAME" [sd-NEW_START_TIME | st-NEW_START_DATE | ed-NEW_END_DATE | et-NEW_END_TIME | n-NEW_NAME | loc-NEW_LOCATION]...
+    edit TASK_UNIQUE_INDEX [sd-NEW_START_TIME | st-NEW_START_DATE | ed-NEW_END_DATE | et-NEW_END_TIME | n-NEW_NAME | loc-NEW_LOCATION]...
```

Comment:
Neither "allow" or "not allow" can be achieved in the parser. But if you think the order should be fixed, then I will modify this part later.